src/intrinsics/ecma262/Object.js

```diff
@@ -157,12 +157,10 @@ export default function(realm: Realm): NativeFunctionValue {
 
       // If we generated an Object.assign() to deal with partials, by this
       // point it is not safe to interact with those objects in Prepack land.
+      Leak.leakValue(to);
       for (let nextSource of sources) {
         Leak.leakValue(realm, nextSource);
```

Comment:
"Leaking" the value seems a bit bizarre. The real issue here is that nextSource is referenced in temporal expression and there needs to be complete and unchanging at that point in time. I guess the overall mechanism is the right one, but the name is misleading.

Reply:
Yea, this is just because we don't have a different mechanism for storing intermediate snapshots that doesn't *also* leak.

Reply:
We could implement a wrapper function for these cases that just leak as an implementation detail for now. That way we can expand on it later if needed.